pkg/proxy/util/nfacct/nfacct_linux.go

```diff
@@ -74,6 +74,13 @@ func newInternal(hndlr handler) (Interface, error) {
 
 }
 
+// Present checks if nfacct is supported on the current kernel by verifying if a counter can be
+// retrieved. Returns true if the call is successful or if it returns ENOENT.
+func (r *runner) Present() bool {
+	_, err := r.Get("IMayExist")
+	return err == nil || errors.Is(err, ErrObjectNotFound)
+}
```

Comment:
Why not just do this in `New()`, and return an error from there?

Reply:
I initially thought about doing that only, but iptables interface has `Present()` method which checks for kernel support and `New()` simply returns the implementation. I followed the same here.
https://github.com/kubernetes/kubernetes/blob/v1.30.1/pkg/util/iptables/iptables.go#L52